server/src/test/java/io/crate/auth/JwtAuthenticationIntegrationTest.java

```diff
@@ -182,6 +238,39 @@ public void test_can_authenticate_with_jwt_token() throws Exception {
                   """);
         }
 
+        assertThat(requestHandler.numberOfInvocation.get()).isEqualTo(1);
+
+        // Now try a second time, same domain different user
+        appUsername = "cloud_user_1";
+        jwt = JWT.create()
+            .withHeader(Map.of("typ", "JWT", "alg", "RS256", "kid", KID))
+            .withIssuer(iss)
+            .withAudience(clusterService().state().metadata().clusterUUID())
+            .withClaim("username", appUsername)
+            .sign(Algorithm.RSA256(null, privateKey));
+
+
+        execute("CREATE USER \"Joe\" " +
+            "WITH (jwt = {\"iss\" = '" + iss + "', \"username\" = '" + appUsername + "'})"
+        );
+
+
+        try (var client = HttpClient.newHttpClient()) {
+            HttpRequest request = HttpRequest.newBuilder(uri)
+                .header(HttpHeaderNames.AUTHORIZATION.toString(), "Bearer " + jwt)
+                .header(HttpHeaderNames.ORIGIN.toString(), "http://example.com")
+                .header(HttpHeaderNames.ACCESS_CONTROL_REQUEST_METHOD.toString(), "GET")
+                .build();
+            var resp = client.send(request, BodyHandlers.ofString());
+            assertThat(resp.body()).containsIgnoringWhitespaces("""
+                {
+                  "ok" : true,
```

Comment:
We could create helper static methods to 
1. fire a request and get response
2. prepare a token with a given app_username

Reply:
sounds good

Reply:
👍